Tasks/AzureRmWebAppDeployment/operations/AzureAppServiceUtility.ts

```diff
@@ -0,0 +1,231 @@
+import tl = require('vsts-task-lib/task');
+import { AzureAppService } from 'azure-arm-rest/azure-arm-app-service';
+import webClient = require('azure-arm-rest/webClient');
+var parseString = require('xml2js').parseString;
+import Q = require('q');
+import { Kudu } from 'azure-arm-rest/azure-arm-app-service-kudu';
+import { AzureAppServiceConfigurationDetails } from 'azure-arm-rest/azureModels';
+
+export class AzureAppServiceUtility {
+    private _appService: AzureAppService;
+    constructor(appService: AzureAppService) {
+        this._appService = appService;
+    }
+
+    public async updateScmTypeAndConfigurationDetails() : Promise<void>{
+            try {
+            var configDetails = await this._appService.getConfiguration();
+            var scmType: string = configDetails.properties.scmType;
+            if (scmType && scmType.toLowerCase() === "none") {
+                configDetails.properties.scmType = 'VSTSRM';
+                tl.debug('updating SCM Type to VSTS-RM');
+                await this._appService.updateConfiguration(configDetails);
+                tl.debug('updated SCM Type to VSTS-RM');
+                tl.debug('Updating metadata with latest release details');
+                await this._appService.patchMetadata(this._getNewMetadata());
+                tl.debug('Updated metadata with latest release details');
+                console.log(tl.loc("SuccessfullyUpdatedAzureRMWebAppConfigDetails"));
+            }
+            else {
+                tl.debug(`Skipped updating the SCM value. Value: ${scmType}`);
+            }
+        }
+        catch(error) {
+            tl.warning(tl.loc("FailedToUpdateAzureRMWebAppConfigDetails", error));
+        }
+    }
+
+    public async getWebDeployPublishingProfile(): Promise<any> {
+        var publishingProfile = await this._appService.getPublishingProfileWithSecrets();
+        var defer = Q.defer<any>();
+        parseString(publishingProfile, (error, result) => {
+            if(result && result.publishData && result.publishData.publishProfile) {
+                for (var index in result.publishData.publishProfile) {
+                    if (result.publishData.publishProfile[index].$ && result.publishData.publishProfile[index].$.publishMethod === "MSDeploy") {
```

Comment:
Suggestion - Instead of result.publishData.publishProfile multiple times should we define a variable.

Reply:
done. thanks